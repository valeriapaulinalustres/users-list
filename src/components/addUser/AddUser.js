import React from 'react';
import './addUser.css';
import Toast from 'sweetalert2';


function AddUser({ onAdd, onEnter, onLeave,  validateName, 
validateMail, invalidName, invalidPhone, invalidEmail, addUser }) {

  //event submit and email, phone and name validations
  const handleOnSubmit = (e) => {
    e.preventDefault();
    console.log(e.target.phone.value)

    const validName = validateName(e.target.name.value);
    if (!validName) return invalidName();

    if (e.target.phone.value == "") return invalidPhone();

    const validEmail = validateMail(e.target.email.value);
    if (!validEmail) return invalidEmail();

    onAdd(e.target.name.value, e.target.phone.value, e.target.email.value);
    e.target.name.value = "";
    e.target.phone.value = "";
    e.target.email.value = "";

    Toast.fire({
      icon: 'success',
      title: 'Added successfully'
    })

  }

 

  return (
    <div className='form-container' role='formContainer'>
      <form onSubmit={handleOnSubmit}>
        <h2>{addUser}</h2>
        <input placeholder="Name" name="name" className='input' type="text" />
        <input placeholder="Phone" name="phone" className='input' type="number" />
        <input placeholder="Email" name="email" className='input' />
        <button 
        onSubmit={handleOnSubmit} 
        className="button" 
        role="buttonSubmit" 
        onMouseEnter={onEnter} 
        onMouseLeave={onLeave}>Add</button>
      </form>
    </div>
  );
}

export default AddUser