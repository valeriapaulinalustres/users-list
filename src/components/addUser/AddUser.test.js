import { fireEvent, render, screen } from '@testing-library/react';
import AddUser from './AddUser';


const addUserProp = "Add User"
  
  it("matches snapshot (PROP)", () => {
    const { asFragment } = render(<AddUser addUserProp={addUserProp} />);
    expect(asFragment()).toMatchSnapshot();
  });

  // describe("AddUser component", () => {
  //   const mockChangeValue = jest.fn();
  //   const stubbedSearchValue = {
  //     name: "Valeria",
  //     phone: "123456",
  //     email: "va@gmail.com",
  //   };
  
  //   it("shows all required input fields with empty values", () => {
  //     const { getByTestId } = render(
  //       <AddUser
  //         searchValue={stubbedSearchValue}
  //         onAdd={mockChangeValue}
  //       />
  //     );
  
  //   //  expect(screen.getAllByPlaceholderText(/name/i).name.value).toBe("Valeria");
  //      expect(getByTestId("filter-input-phone").value).toBe("123456");
  //      expect(getByTestId("filter-input-email").value).toBe("va@gmail.com");
  //   });
  // })

  beforeEach(()=>{
    render(<AddUser />)
})

test('should render form Container', () => {
    const formContainer = screen.getByRole('formContainer');
    expect(formContainer).toBeInTheDocument();
  });

  test('should render input for name', () => {
    const input = screen.getByPlaceholderText(/name/i);
    expect(input).toBeInTheDocument();
  });

  test('should render input for phone', () => {
    const input = screen.getByPlaceholderText(/phone/i);
    expect(input).toBeInTheDocument();
  });

  test('should render input for email', () => {
    const input = screen.getByPlaceholderText(/email/i);
    expect(input).toBeInTheDocument();
  });

  // it('should render a primary button', () => {
  //   const btn = screen.getByText("Add User");
  //   expect(btn).toBeInTheDocument();
  //   expect(btn).toHaveClass('button') 
  // });
