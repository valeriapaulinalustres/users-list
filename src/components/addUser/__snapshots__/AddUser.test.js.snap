// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot (PROP) 1`] = `
<DocumentFragment>
  <div
    class="form-container"
    role="formContainer"
  >
    <form>
      <h2 />
      <input
        class="input"
        data-testid="filter-input-name"
        name="name"
        placeholder="Name"
        type="text"
      />
      <input
        class="input"
        data-testid="filter-input-phone"
        name="phone"
        placeholder="Phone"
        type="number"
      />
      <input
        class="input"
        data-testid="filter-input-email"
        name="email"
        placeholder="Email"
      />
      <button
        class="button"
        role="buttonSubmit"
      >
        Add
      </button>
    </form>
  </div>
</DocumentFragment>
`;
